pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -382,8 +423,13 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 		return false, fmt.Errorf("failed to get cached claims to provision for pod %q", podName)
 	}
 
+	// Check for any conditions that might require scheduling retry
+
 	for _, binding := range bindings {
-		// Check for any conditions that might require scheduling retry
+		// Check if pv is out of date
+		if err := b.pvCache.IsOutOfDate(binding.pv); err != nil {
```

Comment:
This causes lots of unnecessary scheduling retires which increase volume binding latency. TestVolumeBindingDynamicStressFast/TestVolumeBindingDynamicStressSlow have high chance to fail. I'm working on a more fine-grain version.

Reply:
Done